test/e2e/nstemplatetier_test.go

```diff
@@ -0,0 +1,28 @@
+package e2e
+
+import (
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/doubles"
+	"github.com/codeready-toolchain/toolchain-e2e/wait"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateOrUpdateNSTemplateTierAtStartup(t *testing.T) {
+	// given
+	tierList := &toolchainv1alpha1.NSTemplateTierList{}
+	ctx, awaitility := doubles.InitializeOperators(t, tierList)
+	defer ctx.Cleanup()
+	hostAwait := wait.NewHostAwaitility(awaitility)
+
+	// check the "advanced" NSTemplateTier, expecting generation=1 (just created)
+	err := hostAwait.WaitForNSTemplateTier("advanced", wait.NSTemplateTierHavingGeneration(1))
```

Comment:
I'm not sure if checking generation is really a good approach - that is something out of our control, isn't it? Wouldn't be better to only check the existence of the tier in this case?

Reply:
well, there's already an older tier, so I would need to make sure that it was updated first... or should I check the content instead? Eg, make sure that the `namespace[].revision` matches a given value? But since we start with `generation=1`, can't we just assume that `generation=2` is for the updated object?

Reply:
done in 7a4b027